x-pack/plugins/ml/public/services/results_service.js

```diff
@@ -1391,6 +1392,147 @@ function getEventRateData(
   });
 }
 
+// Queries Elasticsearch to obtain event distribution i.e. the count
+// of entities over time.
+// index can be a String, or String[], of index names to search.
+// Extra query object can be supplied, or pass null if no additional query.
+// Returned response contains a results property, which is an object
+// of document counts against time (epoch millis).
+// const SAMPLER_TOP_TERMS_THRESHOLD = 100000;
+const SAMPLER_TOP_TERMS_SHARD_SIZE = 200;
```

Comment:
I wonder if it would make sense to supply this as a parameter to the function, in the same way that is used in the data visualizer functions?

Reply:
Added a note to consider that in the follow-up PR #21163.